### PR TITLE
Create APACHE_PROPOSAL.md

### LINKED ISSUE #1631: [Apache] prepare proposal
### Feature Description (功能描述)

Proposal guide:
http://incubator.apache.org/guides/proposal.html

Historical proposals:
https://cwiki.apache.org/confluence/display/INCUBATOR/Proposals
Such as ShardingSphere proposal:
https://cwiki.apache.org/confluence/display/INCUBATOR/ShardingSphereProposal

ALC Beijing Incubator Cookbook：
https://mp.weixin.qq.com/s/lmMjiYCZMuOBpVwX5UMduA
https://alc-beijing.github.io/alc-site/post/apache-way/incubator-cook-book/
https://alc-beijing.github.io/alc-site/post/apache-way/incubator-cook-book-table/

### PR DESCRIPTION
**Pull request by javeme**
proposal guide: http://incubator.apache.org/guides/proposal.html

close #1631

TODO:

- [x] #1632 
- [ ] #1635

### CODE REVIEW

File: APACHE_PROPOSAL.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# [New Podling Proposal of HugeGraph](https://cwiki.apache.org/confluence/display/INCUBATOR/New+Podling+Proposal)
+
+## Abstract
+
+HugeGraph will be a graph database with high performance and scalability.
+
+## Proposal
+
+HugeGraph is to provide an large-scale graph database, which achieves availability and balance between performance and cost in the scenario of 100+ billion data,  and has complete HTAP capabilities in an internal system.
```
Comment (Linary) [inline APACHE_PROPOSAL.md:9]: "data,  and" has more than one space

**Review by Linary — CHANGES_REQUESTED** [review]

File: APACHE_PROPOSAL.md (hunk 1 above)
Comment (javeme) [reply APACHE_PROPOSAL.md:9]: ok

File: APACHE_PROPOSAL.md [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+
+We have checked that the name is [suitable](https://github.com/hugegraph/hugegraph/issues/1646) and the project has legal permission to continuing using its current name. There is no one else found using this name through Google search.
+
+#### Orphaned products
+
+Due to the relatively small number of contributors, the project has a relatively small risk of becoming an orphan. However, the contributors have been operating in the spirit of open source for 3 years, and continue to develop new contributors to participate.
+
+#### Inexperience with Open Source:
+
+HugeGraph has been open sourced on GitHub for 3 years, during which contributors submitted code and documents in full compliance with open source specifications and requirements
```
Comment (javeme) [inline APACHE_PROPOSAL.md:53]: contributors -- ditto

File: APACHE_PROPOSAL.md [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+
+#### Homogenous Developers:
+
+The current list of committers includes developers from several different companies plus many independent volunteers. The committers are geographically concentrated in China now. They are experienced with working in a distributed environment in other open source project, e.g. OpenStack.
+
+#### Reliance on Salaried Developers:
+
+Most of the developers are paid by their employer to contribute to this project, but given some volunteer developers and the committers' sense of ownership for the code, the project would continue without issue if no salaried developers contributed to the project.
+
+#### Relationship with Titan/Janus Graph
```
Comment (javeme) [inline APACHE_PROPOSAL.md:67]: keep the section in "Known Risks" and add a new section to describe "Relationships with Other Apache Products:"

File: APACHE_PROPOSAL.md [hunk 4]
```diff
@@ -0,0 +1,326 @@
+# [New Podling Proposal of HugeGraph](https://cwiki.apache.org/confluence/display/INCUBATOR/New+Podling+Proposal)
+
+## Abstract
+
+HugeGraph will be a graph database with high performance and scalability.
```
Comment (WillemJiang) [inline APACHE_PROPOSAL.md:5]: will be  -> is

File: APACHE_PROPOSAL.md [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+9. `hugegraph-hubble`: https://github.com/hugegraph/hugegraph-hubble
+10. `hugegraph-test`: https://github.com/hugegraph/hugegraph-test
+
+### Source and Intellectual Property Submission Plan
+
+The codes are currently under Apache License Version 2.0 and have been verified to there is no intellectual property or license issues before being released to open source by Baidu in 2018. Baidu will provide SGA and all committers will sign ICLA after HugeGraph is accepted into the Incubator.
+
+#### External Dependencies:
+
+As all dependencies are managed using Apache Maven, none of the external libraries need to be packaged in a source distribution. All dependencies have Apache compatible licenses except MySQL (GPL-2.0), we will remove MySQL dependencies in future.
```
Comment (WillemJiang) [inline APACHE_PROPOSAL.md:106]: If the MySQL dependency is optional, we can add some instruction for users to download by themselves.

https://www.apache.org/legal/resolved.html#optional

File: APACHE_PROPOSAL.md [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+
+- CDDL + GPLv2 License
+
+  - glassfish
+  - Java Servlet API
+  - javax.ws.rs-api 
+  - jersey-inject-hk2
+  - JSR 353 (JSON Processing) Default Provider
+
+- Commercial License + AGPL License
```
Comment (WillemJiang) [inline APACHE_PROPOSAL.md:215]: We need to take a look at this dependency.  We cannot ship the AGPL third party dependency.
https://www.apache.org/legal/resolved.html#category-x

File: APACHE_PROPOSAL.md [hunk 7]
```diff
@@ -0,0 +1,326 @@
...
+- Eclipse Distribution License
+
+  - Eclipse Collections
+  - JUnit
+
+- ISC/BSD License
+
+  - jBCrypt
+
+- LGPL License
```
Comment (WillemJiang) [inline APACHE_PROPOSAL.md:228]: We need to take a look at this dependency. We cannot ship the GPL and LGPL third party dependency.
https://www.apache.org/legal/resolved.html#category-x

File: APACHE_PROPOSAL.md (hunk 4 above)
Comment (javeme) [reply APACHE_PROPOSAL.md]: done, thanks~

File: APACHE_PROPOSAL.md [hunk 8]
```diff
@@ -0,0 +1,326 @@
...
+
+#### Core Developers
+
+The core developers are already experienced open source developers, they have operated the HugeGraph Community for 3 years, and they are contributors of Linux kernel, OpenStack, Ceph, RocksDB, Apache TinkerPop, Apache Hadoop and Apache Groovy.
+
+#### Alignment
+
+HugeGraph is compliance to [Apache TinkerPop 3](https://tinkerpop.apache.org/) framework, various complicated graph queries can be accomplished through [Gremlin](https://tinkerpop.apache.org/gremlin.html), a powerful graph traversal language. The TinkerPop ecosystem based on Java is relatively mature in terms of graph databases, we use Java language which is develop efficient and stable to develop upper-level components like graph engine, graph computing, graph api and graph tools; and we manage storage through JNI which is able to freely manage memory and execute efficiently.
```
Comment (WillemJiang) [inline APACHE_PROPOSAL.md:41]: compliance 的翻译有点问题， 是说我们在用TinkerPop API吗？

**Review by WillemJiang — COMMENTED** [review]
Just add some comments on this proposal.

File: APACHE_PROPOSAL.md [hunk 9]
```diff
@@ -0,0 +1,326 @@
...
+5. `hugegraph-doc`: https://github.com/hugegraph/hugegraph-doc.git
+6. `hugegraph-loader`: https://github.com/hugegraph/hugegraph-loader.git
+7. `hugegraph-tools`: https://github.com/hugegraph/hugegraph-tools.git
+8. `hugegraph-rpc`: https://github.com/hugegraph/hugegraph-rpc.git
+9. `hugegraph-hubble`: https://github.com/hugegraph/hugegraph-hubble.git
+10. `hugegraph-test`: https://github.com/hugegraph/hugegraph-test.git
+
+#### Issue Tracking:
+
+We choose JIRA HugeGraph (HUGEGRAPH), and the community would like to continue using [GitHub Issues](https://github.com/hugegraph/hugegraph/issues).
```
[the remark this answers is not in the document]
Comment (javeme) [reply APACHE_PROPOSAL.md:289]: Ok, we prefer to still use GitHub Issues

File: APACHE_PROPOSAL.md (hunk 5 above)
Comment (javeme) [reply APACHE_PROPOSAL.md:106]: Yes, the MySQL dependency is optional.
TODO: add some instruction for users to download by themselves

File: APACHE_PROPOSAL.md (hunk 6 above)
Comment (javeme) [reply APACHE_PROPOSAL.md:215]: TODO: check AGPL dependency

File: APACHE_PROPOSAL.md (hunk 7 above)
Comment (javeme) [reply APACHE_PROPOSAL.md:228]: TODO: check LGPL dependency

File: APACHE_PROPOSAL.md (hunk 8 above)
Comment (javeme) [reply APACHE_PROPOSAL.md:41]: 想要表达的是” HugeGraph遵循TinkerPop的规范“，比如”完全兼容TinkerPop定义的Gremlin查询语言“

**Review by coderzc — APPROVED** [review]

**Review by corgiboygsj — APPROVED** [review]

**Pull request merged**